modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/dml/UpdatePlan.java

```diff
@@ -531,4 +534,81 @@ public boolean isLocalSubquery() {
     public boolean canSelectBeLazy() {
         return canSelectBeLazy;
     }
+
+    /**
+     * @return String representation of update plan.
+     */
+    public String plan() {
+        StringBuilder sb = new StringBuilder("mode=" + mode +
+            "; table name=" + tbl.cacheName() +
+            "; group ID=" + tbl.cacheContext().groupId() +
+            "; columns=" + (colNames == null ? "N/A" : "[" + Arrays.stream(colNames).reduce((a, b) -> a + ", " + b)
+            .orElse("") + "]") +
+            "; column types=" + (colTypes == null ? "N/A" : "[" + Arrays.stream(colTypes).mapToObj(Objects::toString)
+            .reduce((a, b) -> a + ", " + b).orElse("") + "]") +
+            "; key column index=" + (keyColIdx < 0 ? "N/A" : keyColIdx) +
+            "; value column index=" + (valColIdx < 0 ? "N/A" : valColIdx) +
+            "; row count=" + rowsNum +
+            "; values=" + (rows == null ? "N/A" : "[" + rowsToString(rows) + "]") +
```

Comment:
Not sure we should add rows to plan. Plan will be different each time and it will wash out all other plan. For me value of DML plans are not clear. Perhaps it will be useful to have only plan for selectQry, but looks like we don't have this plan during update. Perhaps these queries are tracked on map phase.